File: scripts/seed-db.js
const { PrismaClient } = require('@prisma/client');

const prisma = new PrismaClient();

async function main() {
  // Create demo user
  const userPassword = '$2a$12$wHjp1RYkQHyPuME.lmF0XuVwD3b/pX47ROfL4sOhW5IHbIL3PKdqy'; // user123
  
  try {
    await prisma.user.create({
      data: {
        id: 'user-001',
        email: 'user@pachmarhi.com',
        password: userPassword,
        name: 'Demo User',
        role: 'USER',
        isActive: true,
        emailVerified: true
      }
    });
    console.log('Created demo user');
  } catch (error) {
    console.log('Demo user already exists');
  }

  // Create categories
  const categories = [
    {
      id: 'tribal-shirts',
      name: { en: 'Tribal Shirts', hi: 'जनजातीय शर्ट' },
      slug: 'tribal-shirts',
      description: { 
        en: 'Traditional tribal printed shirts and ethnic wear', 
        hi: 'पारंपरिक जनजातीय मुद्रित शर्ट और जातीय पहनावा' 
      },
      image: '/images/categories/cat-tribal-shirts.jpg',
      featured: true,
      isActive: true,
      productCount: 0,
      sortOrder: 1
    },
    {
      id: 'jewelry',
      name: { en: 'Jewelry', hi: 'आभूषण' },
      slug: 'jewelry',
      description: { 
        en: 'Handcrafted tribal jewelry and accessories', 
        hi: 'हस्तनिर्मित जनजातीय आभूषण और सहायक उपकरण' 
      },
      image: '/images/categories/cat-jewelry.jpg',
      featured: true,
      isActive: true,
      productCount: 0,
      sortOrder: 2
    },
    {
      id: 'handloom-textiles',
      name: { en: 'Handloom Textiles', hi: 'हैंडलूम वस्त्र' },
      slug: 'handloom-textiles',
      description: { 
        en: 'Traditional handwoven textiles and fabrics', 
        hi: 'पारंपरिक हस्तबुने वस्त्र और कपड़े' 
      },
      image: '/images/categories/cat-handloom-textiles.jpg',
      featured: true,
      isActive: true,
      productCount: 0,
      sortOrder: 3
    },
    {
      id: 'home-decor',
      name: { en: 'Home Decor', hi: 'घर की सजावट' },
      slug: 'home-decor',
      description: { 
        en: 'Beautiful home decor items and tribal art', 
        hi: 'सुंदर घर की सजावट की वस्तुएं और जनजातीय कला' 
      },
      image: '/images/categories/cat-home-decor.jpg',
      featured: true,
      isActive: true,
      productCount: 0,
      sortOrder: 4
    },
    {
      id: 'accessories',
      name: { en: 'Accessories', hi: 'सहायक उपकरण' },
      slug: 'accessories',
      description: { 
        en: 'Tribal accessories and decorative items', 
        hi: 'जनजातीय सहायक उपकरण और सजावटी वस्तुएं' 
      },
      image: '/images/categories/cat-accessories.jpg',
      featured: false,
      isActive: true,
      productCount: 0,
      sortOrder: 5
    },
    {
      id: 'gifts-souvenirs',
      name: { en: 'Gifts & Souvenirs', hi: 'उपहार और स्मृति चिन्ह' },
      slug: 'gifts-souvenirs',
      description: { 
        en: 'Perfect gifts and memorable souvenirs', 
        hi: 'पूर्ण उपहार और यादगार स्मृति चिन्ह' 
      },
      image: '/images/categories/cat-gifts-souvenirs.jpg',
      featured: false,
      isActive: true,
      productCount: 0,
      sortOrder: 6
    }
  ];

  for (const categoryData of categories) {
    try {
      await prisma.category.create({
        data: categoryData
      });
      console.log(`Created category: ${categoryData.name.en}`);
    } catch (error) {
      console.log(`Category already exists: ${categoryData.name.en}`);
    }
  }

  // Create artisans
  const artisans = [
    {
      id: 'sarla-bai',
      name: 'Sarla Bai',
      slug: 'sarla-bai',
      bio: { 
        en: 'Master weaver from Pachmarhi with 25 years of experience in traditional handloom techniques.', 
        hi: 'पचमढ़ी की मास्टर बुनकर जिनके पास पारंपरिक हैंडलूम तकनीकों में 25 साल का अनुभव है।' 
      },
      village: 'Pachmarhi',
      district: 'Hoshangabad',
      state: 'Madhya Pradesh',
      photo: '/images/artisans/arti-sarla.jpg',
      experience: 25,
      rating: 4.8,
      totalProducts: 15,
      isActive: true,
      joinedAt: new Date()
    },
    {
      id: 'ramesh-uikey',
      name: 'Ramesh Uikey',
      slug: 'ramesh-uikey',
      bio: { 
        en: 'Skilled pottery artisan specializing in terracotta crafts.', 
        hi: 'टेराकोटा शिल्प में विशेषज्ञता रखने वाले कुशल मिट्टी के बर्तन कारीगर।' 
      },
      village: 'Bori',
      district: 'Hoshangabad',
      state: 'Madhya Pradesh',
      photo: '/images/artisans/arti-ramesh.jpg',
      experience: 18,
      rating: 4.6,
      totalProducts: 12,
      isActive: true,
      joinedAt: new Date()
    },
    {
      id: 'meera-gond',
      name: 'Meera Gond',
      slug: 'meera-gond',
      bio: { 
        en: 'Renowned Gond painting artist preserving ancient tribal art forms.', 
        hi: 'प्राचीन जनजातीय कला रूपों को संरक्षित करने वाली प्रसिद्ध गोंड चित्रकारी कलाकार।' 
      },
      village: 'Satpura',
      district: 'Hoshangabad',
      state: 'Madhya Pradesh',
      photo: '/images/artisans/arti-meera.jpg',
      experience: 22,
      rating: 4.9,
      totalProducts: 20,
      isActive: true,
      joinedAt: new Date()
    }
  ];

  for (const artisanData of artisans) {
    try {
      await prisma.artisan.create({
        data: artisanData
      });
      console.log(`Created artisan: ${artisanData.name}`);
    } catch (error) {
      console.log(`Artisan already exists: ${artisanData.name}`);
    }
  }

  // Create products
  const products = [
    {
      id: 'bamboo-wall-art',
      title: { en: 'Bamboo Wall Art', hi: 'बांस की दीवार कला' },
      slug: 'bamboo-wall-art',
      description: { 
        en: 'Beautiful handcrafted bamboo wall art featuring traditional tribal motifs.', 
        hi: 'पारंपरिक जनजातीय रूपांकनों को दर्शाने वाली सुंदर हस्तनिर्मित बांस की दीवार कला।' 
      },
      shortDesc: { 
        en: 'Handcrafted bamboo wall art', 
        hi: 'हस्तनिर्मित बांस की दीवार कला' 
      },
      price: 1699.00,
      originalPrice: 1899.00,
      sku: 'BWA-001',
      stock: 12,
      tags: '["wall art", "bamboo", "eco-friendly", "tribal", "home decor"]',
      isActive: true,
      inStock: true,
      featured: true,
      bestSeller: true,
      newArrival: false,
      trending: false,
      rating: 4.5,
      reviewCount: 24,
      viewCount: 0,
      saleCount: 0,
      categoryId: 'home-decor',
      artisanId: 'ramesh-uikey'
    },
    {
      id: 'handloom-sari',
      title: { en: 'Handloom Sari', hi: 'हैंडलूम साड़ी' },
      slug: 'handloom-sari',
      description: { 
        en: 'Exquisite handwoven sari featuring traditional tribal patterns.', 
        hi: 'पारंपरिक जनजातीय पैटर्न को दर्शाने वाली उत्कृष्ट हस्तबुनी साड़ी।' 
      },
      shortDesc: { 
        en: 'Exquisite handwoven sari', 
        hi: 'उत्कृष्ट हस्तबुनी साड़ी' 
      },
      price: 2999.00,
      originalPrice: 3499.00,
      sku: 'HLS-001',
      stock: 8,
      tags: '["sari", "handloom", "cotton", "traditional", "ethnic wear"]',
      isActive: true,
      inStock: true,
      featured: true,
      bestSeller: true,
      newArrival: false,
      trending: false,
      rating: 4.8,
      reviewCount: 36,
      viewCount: 0,
      saleCount: 0,
      categoryId: 'handloom-textiles',
      artisanId: 'sarla-bai'
    },
    {
      id: 'tribal-silver-earrings',
      title: { en: 'Tribal Silver Earrings', hi: 'जनजातीय चांदी के कान के झूले' },
      slug: 'tribal-silver-earrings',
      description: { 
        en: 'Handcrafted silver earrings with traditional tribal designs.', 
        hi: 'पारंपरिक जनजातीय डिज़ाइनों के साथ हस्तनिर्मित चांदी के कान के झूले।' 
      },
      shortDesc: { 
        en: 'Handcrafted silver earrings', 
        hi: 'हस्तनिर्मित चांदी के कान के झूले' 
      },
      price: 899.00,
      originalPrice: 999.00,
      sku: 'TSE-001',
      stock: 25,
      tags: '["earrings", "silver", "tribal", "jewelry", "accessories"]',
      isActive: true,
      inStock: true,
      featured: true,
      bestSeller: false,
      newArrival: false,
      trending: true,
      rating: 4.6,
      reviewCount: 18,
      viewCount: 0,
      saleCount: 0,
      categoryId: 'jewelry',
      artisanId: 'ramesh-uikey'
    }
  ];

  for (const productData of products) {
    try {
      await prisma.product.create({
        data: {
          ...productData,
          productImages: {
            create: [
              {
                url: `/images/products/${productData.slug}/img1.jpg`,
                isPrimary: true,
                sortOrder: 0
              },
              {
                url: `/images/products/${productData.slug}/img2.jpg`,
                isPrimary: false,
                sortOrder: 1
              },
              {
                url: `/images/products/${productData.slug}/img3.jpg`,
                isPrimary: false,
                sortOrder: 2
              }
            ]
          }
        }
      });
      console.log(`Created product: ${productData.title.en}`);
    } catch (error) {
      console.log(`Product already exists: ${productData.title.en}`);
    }
  }

  console.log('Database seeding completed!');
}

main()
  .catch((e) => {
    console.error(e);
    process.exit(1);
  })
  .finally(async () => {
    await prisma.$disconnect();
  });

File: seed-db.py
import mysql.connector
import bcrypt
from datetime import datetime, timedelta

# Database connection
conn = mysql.connector.connect(
    host='localhost',
    user='root',
    password='',
    database='pachmarhi_db'
)
cursor = conn.cursor()

print("🌱 Starting database seed...")

try:
    # Clear existing data
    print("🧹 Cleaning existing data...")
    cursor.execute("DELETE FROM coupon_usage")
    cursor.execute("DELETE FROM coupons")
    cursor.execute("DELETE FROM banners")
    cursor.execute("DELETE FROM product_reviews")
    cursor.execute("DELETE FROM returns")
    cursor.execute("DELETE FROM order_items")
    cursor.execute("DELETE FROM orders")
    cursor.execute("DELETE FROM wishlists")
    cursor.execute("DELETE FROM carts")
    cursor.execute("DELETE FROM product_variants")
    cursor.execute("DELETE FROM product_images")
    cursor.execute("DELETE FROM products")
    cursor.execute("DELETE FROM artisans")
    cursor.execute("DELETE FROM categories")
    cursor.execute("DELETE FROM user_addresses")
    cursor.execute("DELETE FROM refresh_tokens")
    cursor.execute("DELETE FROM users WHERE email NOT IN ('admin@pachmarhi.com')")

    # Hash passwords
    hashed_admin_password = bcrypt.hashpw('admin123'.encode('utf-8'), bcrypt.gensalt())
    hashed_user_password = bcrypt.hashpw('user123'.encode('utf-8'), bcrypt.gensalt())

    # Create Users
    print("👤 Creating users...")
    cursor.execute(
        "INSERT INTO users (id, email, password_hash, name, role, email_verified) VALUES (%s, %s, %s, %s, %s, %s)",
        ('admin-001', 'admin@pachmarhi.com', hashed_admin_password, 'Admin User', 'super_admin', True)
    )
    
    cursor.execute(
        "INSERT INTO users (id, email, password_hash, name, role, email_verified) VALUES (%s, %s, %s, %s, %s, %s)",
        ('user-001', 'user@pachmarhi.com', hashed_user_password, 'Demo User', 'customer', True)
    )

    # Create Categories
    print("📂 Creating categories...")
    categories = [
        ('tribal-shirts', 'Tribal Shirts', 'जनजातीय शर्ट', 'Traditional tribal printed shirts and ethnic wear', 'पारंपरिक जनजातीय मुद्रित शर्ट और जातीय पहनावा', '/images/categories/cat-tribal-shirts.jpg', 1),
        ('jewelry', 'Jewelry', 'आभूषण', 'Handcrafted tribal jewelry and accessories', 'हस्तनिर्मित जनजातीय आभूषण और सहायक उपकरण', '/images/categories/cat-jewelry.jpg', 2),
        ('handloom-textiles', 'Handloom Textiles', 'हैंडलूम वस्त्र', 'Traditional handwoven textiles and fabrics', 'पारंपरिक हस्तबुने वस्त्र और कपड़े', '/images/categories/cat-handloom-textiles.jpg', 3),
        ('home-decor', 'Home Decor', 'घर की सजावट', 'Beautiful home decor items and tribal art', 'सुंदर घर की सजावट की वस्तुएं और जनजातीय कला', '/images/categories/cat-home-decor.jpg', 4),
        ('accessories', 'Accessories', 'सहायक उपकरण', 'Tribal accessories and decorative items', 'जनजातीय सहायक उपकरण और सजावटी वस्तुएं', '/images/categories/cat-accessories.jpg', 5),
        ('gifts-souvenirs', 'Gifts & Souvenirs', 'उपहार और स्मृति चिन्ह', 'Perfect gifts and memorable souvenirs', 'पूर्ण उपहार और यादगार स्मृति चिन्ह', '/images/categories/cat-gifts-souvenirs.jpg', 6)
    ]

    cursor.executemany(
        "INSERT INTO categories (id, name_en, name_hi, description_en, description_hi, image, display_order) VALUES (%s, %s, %s, %s, %s, %s, %s)",
        categories
    )

    # Create Artisans
    print("👨‍🎨 Creating artisans...")
    artisans = [
        ('sarla-bai', 'sarla-bai', 'Sarla Bai', 'Master weaver from Pachmarhi with 25 years of experience in traditional handloom techniques.', 'पचमढ़ी की मास्टर बुनकर जिनके पास पारंपरिक हैंडलूम तकनीकों में 25 साल का अनुभव है।', 'Pachmarhi', 'Hoshangabad', 'Madhya Pradesh', '/images/artisans/arti-sarla.jpg', '["Handloom Weaving", "Traditional Textiles"]', 25, 4.8, 15),
        ('ramesh-uikey', 'ramesh-uikey', 'Ramesh Uikey', 'Skilled pottery artisan specializing in terracotta crafts.', 'टेराकोटा शिल्प में विशेषज्ञता रखने वाले कुशल मिट्टी के बर्तन कारीगर।', 'Bori', 'Hoshangabad', 'Madhya Pradesh', '/images/artisans/arti-ramesh.jpg', '["Pottery", "Terracotta", "Ceramics"]', 18, 4.6, 12),
        ('meera-gond', 'meera-gond', 'Meera Gond', 'Renowned Gond painting artist preserving ancient tribal art forms.', 'प्राचीन जनजातीय कला रूपों को संरक्षित करने वाली प्रसिद्ध गोंड चित्रकारी कलाकार।', 'Satpura', 'Hoshangabad', 'Madhya Pradesh', '/images/artisans/arti-meera.jpg', '["Gond Painting", "Traditional Art", "Wall Art"]', 22, 4.9, 20),
        ('raj-kumar', 'raj-kumar', 'Raj Kumar', 'Expert bamboo and cane craftsman creating sustainable products.', 'बांस और बेंत के विशेषज्ञ कारीगर जो टिकाऊ उत्पाद बनाते हैं।', 'Pachmarhi', 'Hoshangabad', 'Madhya Pradesh', '/images/artisans/arti-raj.jpg', '["Bamboo Craft", "Cane Work", "Eco Products"]', 20, 4.7, 18)
    ]

    cursor.executemany(
        "INSERT INTO artisans (id, user_id, name, bio_en, bio_hi, specialization, location, phone, email, avatar, portfolio_images, social_links, experience_years, rating, total_products, is_verified, is_active) VALUES (%s, %s, %s, %s, %s, NULL, %s, %s, NULL, %s, %s, NULL, %s, %s, %s, TRUE, TRUE)",
        artisans
    )

    # Create Products
    print("🛍️ Creating products...")
    products = [
        ('bamboo-wall-art', 'Bamboo Wall Art', 'बांस की दीवार कला', 'Beautiful handcrafted bamboo wall art featuring traditional tribal motifs.', 'पारंपरिक जनजातीय रूपांकनों को दर्शाने वाली सुंदर हस्तनिर्मित बांस की दीवार कला।', 1699.00, 1899.00, 10, 'BWA-001', 12, '["Bamboo"]', '["wall art", "bamboo", "eco-friendly", "tribal", "home decor"]', True, True, False, False, 4.5, 24, 0, 0, 'home-decor', 'raj-kumar'),
        ('handloom-sari', 'Handloom Sari', 'हैंडलूम साड़ी', 'Exquisite handwoven sari featuring traditional tribal patterns.', 'पारंपरिक जनजातीय पैटर्न को दर्शाने वाली उत्कृष्ट हस्तबुनी साड़ी।', 2999.00, 3499.00, 15, 'HLS-001', 8, '["Cotton"]', '["sari", "handloom", "cotton", "traditional", "ethnic wear"]', True, True, False, False, 4.8, 36, 0, 0, 'handloom-textiles', 'sarla-bai'),
        ('tribal-silver-earrings', 'Tribal Silver Earrings', 'जनजातीय चांदी के कान के झूले', 'Handcrafted silver earrings with traditional tribal designs.', 'पारंपरिक जनजातीय डिज़ाइनों के साथ हस्तनिर्मित चांदी के कान के झूले।', 899.00, 999.00, 10, 'TSE-001', 25, '["Silver"]', '["earrings", "silver", "tribal", "jewelry", "accessories"]', True, False, True, False, 4.6, 18, 0, 0, 'jewelry', 'ramesh-uikey'),
        ('gond-painting', 'Gond Painting', 'गोंड चित्रकारी', 'Authentic Gond painting depicting tribal life and nature.', 'जनजातीय जीवन और प्रकृति को चित्रित करने वाली प्रामाणिक गोंड चित्रकारी।', 3499.00, 3999.00, 10, 'GP-001', 5, '["Canvas", "Acrylic Paint"]', '["painting", "gond", "tribal", "art", "wall decor"]', True, False, False, True, 4.9, 15, 0, 0, 'art-paintings', 'meera-gond'),
        ('tribal-printed-shirt', 'Tribal Printed Shirt', 'जनजातीय मुद्रित शर्ट', 'Comfortable cotton shirt with authentic tribal prints.', 'प्रामाणिक जनजातीय प्रिंट के साथ आरामदायक कपास की शर्ट।', 1299.00, 1499.00, 10, 'TPS-001', 20, '["Cotton"]', '["shirt", "tribal", "cotton", "clothing", "ethnic wear"]', False, False, True, False, 4.4, 32, 0, 0, 'tribal-shirts', 'sarla-bai')
    ]

    cursor.executemany(
        "INSERT INTO products (slug, title_en, title_hi, description_en, description_hi, price, original_price, discount_percentage, sku, stock, material, tags, featured, best_seller, trending, new_arrival, rating, review_count, view_count, sales_count, category_id, artisan_id) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)",
        products
    )

    # Create Banners
    print("🖼️ Creating banners...")
    banners = [
        ('Handmade Bamboo Crafts', 'हस्तनिर्मित बाँस शिल्प', 'Sustainable & Beautiful', 'टिकाऊ और सुंदर', '/images/hero/hero1.jpg', '/category/home-decor', 'HERO', True, 1),
        ('Traditional Textiles', 'पारंपरिक वस्त्र', 'Handwoven with Love', 'प्यार से हाथ से बुने', '/images/hero/hero2.jpg', '/category/handloom-textiles', 'HERO', True, 2),
        ('Tribal Jewelry Collection', 'जनजातीय आभूषण संग्रह', 'Authentic & Unique', 'प्रामाणिक और अद्वितीय', '/images/hero/hero3.jpg', '/category/jewelry', 'HERO', True, 3)
    ]

    cursor.executemany(
        "INSERT INTO banners (title_en, title_hi, subtitle_en, subtitle_hi, image_desktop, link_url, type, is_active, display_order) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s)",
        banners
    )

    # Create Coupons
    print("🎫 Creating coupons...")
    now = datetime.now()
    thirty_days = now + timedelta(days=30)
    fifteen_days = now + timedelta(days=15)
    sixty_days = now + timedelta(days=60)

    coupons = [
        ('WELCOME10', 'Welcome Discount', '10% off on your first order', 'percentage', 10.00, 500.00, 500.00, 100, 0, 1, now, thirty_days, None, None, True),
        ('FESTIVE50', 'Festival Offer', '₹50 off on orders above ₹1000', 'fixed', 50.00, 1000.00, None, 200, 0, 1, now, fifteen_days, None, None, True),
        ('FREESHIP', 'Free Shipping', 'Free shipping on orders above ₹999', 'free_shipping', 0.00, 999.00, None, 500, 0, 1, now, sixty_days, None, None, True)
    ]

    cursor.executemany(
        "INSERT INTO coupons (code, title, description, type, value, minimum_order_amount, maximum_discount_amount, usage_limit, usage_count, user_usage_limit, valid_from, valid_until, applicable_categories, applicable_products, is_active) VALUES (%s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s, %s)",
        coupons
    )

    # Commit changes
    conn.commit()
    print("✅ Database seeded successfully!")
    print("👤 Admin User: admin@pachmarhi.com / admin123")
    print("👤 Demo User: user@pachmarhi.com / user123")
    
except Exception as e:
    print(f"❌ Seed failed: {e}")
    conn.rollback()
finally:
    cursor.close()
    conn.close()